crates/meilisearch/src/search/federated/perform.rs

```diff
@@ -59,14 +61,22 @@ pub async fn perform_federated_search(
 
     // 1. partition queries by host and index
     let mut partitioned_queries = PartitionedQueries::new();
+
+    // Store the original queries order for later metadata building
+    let original_queries = queries.clone();
```

Comment:
I'd rather we avoid this clone if possible (queries can be large), and work to populate the metadata from the results of the individual search queries.